pandas/core/groupby/ops.py

```diff
@@ -833,6 +833,14 @@ def levels(self):
     def names(self):
         return [self.binlabels.name]
 
+    @property
+    def recons_labels(self):
+        comp_ids, obs_ids, _ = self.group_info
+        if len(self.binlabels) != len(self.indices):
```

Comment:
Is this check not appropriate for the BaseGrouper class (disclaimer: I'm not familiar with groupby)? I'm a bit concerned about the amount of code duplication here.

Reply:
This issue occurs when only `groupby` with `freq` for now. and its grouper is `BinGrouper`. I was also able to use just the LBYL way to check whether this is BaseGrouper or BinGrouper. Because I a little bit worried about overhead to check instance type, I took overriding way. However, code duplication more matters, I'll move this logic into BaseGrouper. I don't know whose priority is higher, so let me know it, please.

Reply:
@WillAyd do you have thoughts here?

Reply:
So IIUC the issue here is that the length of the grouper exceeds the length of the object being grouped? Seems a little strange to just return right here but some context would help clarify

Reply:
> So IIUC the issue here is that the length of the grouper exceeds the length of the object being grouped? Seems a little strange to just return right here but some context would help clarify

Yes, exactly, the issue occurs when each length of the label list and the indices(Its key has grouped index and value has values in each group). However, I'm not sure where this logic should be placed. I always appreciate any suggestions for this.

Reply:
this type of change should be in decons_obs_group_ids; having this here is duplicative. This is pretty tricky code and the last thing we want is more of it.

Reply:
@jreback I moved the logic from `recons_labels` to `decons_obs_group_ids`. And I modified it to work in `decons_obs_group_ids` by checking `shape` and the length of unique `comp_ids` and `labels`. I tried not to add new parameters that check the instance type instead, I make the most of the variables in that context.

Reply:
@jreback I think a modification should be under the method of `BinGrouper` because this kind of issue occurs only with `groupby` having `freq`. And if it has `freq`, then it would be `BinGrouper`. I could somewhat manage to do it in `decons_obs_group_ids` though it's quite tricky as you see.  In the first place, if the type of grouper is BinGrouper, then it doesn't need both `labels` and `decons_obs_group_ids` and It's ok to just return unique comp_ids(checked by `pytest pandas` but not absolutely sure).

However why unique comp_ids? that's because `rep`, a partial function which is used at making result index expects unique ids that would be result index. `rep` seems to do its job well. And we know we should change something about `self.recons_labels`.

If you don't like current PR(tricky length checking), consider the overriding again, please. Thank you

Reply:
My suggestion is this:
```diff
diff --git a/pandas/core/groupby/ops.py b/pandas/core/groupby/ops.py
@@ -792,6 +792,11 @@ class BinGrouper(BaseGrouper):
+    @cache_readonly
+    def recons_labels(self):
+        comp_ids, _, _ = self.group_info
+        return [np.unique(comp_ids)]
+
diff --git a/pandas/core/sorting.py b/pandas/core/sorting.py
@@ -162,10 +162,6 @@ def decons_obs_group_ids(comp_ids, obs_ids, shape, labels, xnull):
-    labels = list(labels)
-    unique_comp_ids = np.unique(comp_ids)
-    if (shape[0] != len(unique_comp_ids)) and (shape[0] == len(labels[0])):
-        return [unique_comp_ids]
```

I confirmed it passes all tests. If you don't mind, I'll push it.

Reply:
@jreback any comment, please?